### ISSUE TITLE
byteToBinary() should be removed

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1220]
In writing code for #1219 I realized that R provides a very nice function called `rawToBits` that does something similar to `byteToBinary(x, "big")` (except that the results are given in a vector, and that the least-significant bit is given first).

Therefore, I am going to deprecate `byteToBinary`. There seems no point in duplicating something that's in the "base" package, since any user who is likely to need such things will already be familiar with `rawToBits`. Also, I'd like users who are *not* familiar to see that `rawToBits` is part of a family of useful functions. I don't know why I never ran across these.

The work won't be too hard since

      $ git grep byteToBinary | wc
           24     196    2249
but I'll be sure to take it slowly, to avoid e.g. breaking an orientation in an `adp` file that is interpreted by a two-bit sequence.  (I will also add elements to the test suite to prevent introducing errors ... that's going to be 2/3 of the work.)

I think the only problem will be if I was using little-endian, because `byteToBinary` swaps nibbles, which I think is **wrong**; see

     $ git grep byteToBinary | grep little
     adp.nortek.R:            head$config <- byteToBinary(buf[o+5:6], endian="little")
     adp.nortek.R:            user$mode <- byteToBinary(buf[o+59:60], endian="little")

PS. I half think that there are some `adp` oddnesses from the past that might be related to bits, and as I recode things, I'll be on the lookout for them. 

**checklist** bold-face is used for ones that may be erroneous now because `endian="little"` was used, and I think that might be giving wrong results (but I may already have compensated for it...)

* [ ] **adp.nortek.R:113:**            head$config <- byteToBinary(buf[o+5:6], endian="little")
* [ ] **adp.nortek.R:188:**            user$mode <- byteToBinary(buf[o+59:60], endian="little")
* [ ] adp.rdi.R:128:    systemConfiguration <- paste(byteToBinary(FLD[5], endian="big"), byteToBinary(FLD[6], endian="big"), sep="-")
* [ ] adp.rdi.R:129:    oceDebug(debug, "FLD[4]=", byteToBinary(FLD[4], endian="big"), "(looking near the systemConfiguration bytes to find a problem)\n")
* [ ] adp.rdi.R:130:    oceDebug(debug, "FLD[5]=", byteToBinary(FLD[5], endian="big"), "(should be one of the systemConfiguration bytes)\n")
* [ ] adp.rdi.R:131:    oceDebug(debug, "FLD[6]=", byteToBinary(FLD[6], endian="big"), "(should be one of the systemConfiguration bytes)\n")
* [ ] adp.rdi.R:132:    oceDebug(debug, "FLD[7]=", byteToBinary(FLD[7], endian="big"), "(looking near the systemConfiguration bytes to find a problem)\n")
* [ ] adp.rdi.R:212:    coordTransform <- byteToBinary(FLD[26], endian="big")
* [ ] adp.rdi.R:213:    bits <- substr(byteToBinary(FLD[26], endian="big"), 4, 5)    
* [ ] adp.rdi.R:219:    bits <- substr(byteToBinary(FLD[26], endian="big"), 6, 6)
* [ ] adp.rdi.R:222:    bits <- substr(byteToBinary(FLD[26], endian="big"), 7, 7)
* [ ] adp.rdi.R:225:    bits <- substr(byteToBinary(FLD[26], endian="big"), 8, 8)
* [ ] adp.rdi.R:231:    sensorSource <- byteToBinary(FLD[31], endian="big")
* [ ] adp.rdi.R:232:    sensorsAvailable<- byteToBinary(FLD[32], endian="big")
* [ ] adv.nortek.R:323:    res@metadata$velocityScale <- if ("0" == substr(byteToBinary(buf[vsdStart[1] + 23], endian="big"), 7, 7)) 1e-3 else 0.1e-3
* [ ] adv.nortek.R:326:              byteToBinary(buf[vsdStart[1] + 23], endian="big"), ")\n")
* [ ] adv.nortek.R:509:    res@metadata$orientation <- if ("0" == substr(byteToBinary(status, endian="big"), 1, 1)) "upward" else "downward"

**Comment by dankelley** [comment]
On second thought, after some more thought and inspection of various oce functions, I have decided to *keep* `byteToBinary`, but to simply disallow the `endian="little"` option. That means just two spots of code need alteration, both in `adp.nortek`. 

The reason I changed my mind is that it is quite a lot easier dealing with the output of `byteToBinary`, e.g. the following is a lot clearer than checking the elements of a vector for whether they equal `0x01` or `0x00`, and I think it's also better than doing the binary conversions (i.e. it's better than asking anyone reading the code to know that 011 binary is 3 decimal)
```R
if (bits == "010") frequency <- 250        # kHz
else if (bits == "011") frequency <-  500
else if (bits == "100") frequency <-  1000
else stop("unknown freq. bit code:", bits, " (expect 010 for 250kHz, 010 for 500kHz, etc)")
```

**Comment by dankelley** [comment]
New checklist
```R
git grep byteToBinary|grep -v "big"
```
* [ ] adp.nortek.R:            head$config <- byteToBinary(buf[o+5:6], endian="little")
* [ ] adp.nortek.R:            user$mode <- byteToBinary(buf[o+59:60], endian="little")

**Comment by dankelley** [comment]
I will be testing my recoding of `adp.nortek.R` using `/data/archive/sleiwex/2008/moorings/m05/adp/nortek_aqd2843/raw/adp_nortek_aqd2843.prf` as a test file. Q for @richardsc -- do we have other Nortek adp files in sleiwex?

**Comment by dankelley** [comment]
This comment will hold my notes on the `adp.nortek.R:113` case ... it will be edited in-place, to avoid driving other developers nuts with email notifications.

**Comment by dankelley** [comment]
I feel that this is all fixed now in `develop` commit 677c8332ff0f706dadbb4de0b344eb28895191cf; below is a test code and the summary and graph it makes.
```R
library(oce)
d <- read.oce("/data/archive/sleiwex/2008/moorings/m05/adp/nortek_aqd2843/raw/adp_nortek_aqd2843.prf")
summary(d)
plot(d)
```
The output
```R
ADP Summary
-----------

* Instrument:         aquadoppHR
* Serial number:      AQD 2843
* Firmware version:   1.06
* Source filename:    ``/data/archive/sleiwex/2008/moorings/m05/adp/nortek_aqd2843/raw/adp_nortek_aqd2843.prf``
* Location:           unknown latitude, unknown longitude
* Number of profiles: 101230 
* Number of cells:    25 
* Number of beams:    3 
* Cell size:          0.04000322 m
* Cells:              25, centered at 0.092 m to 1.052 m, spaced by 0.040 m
* Coordinate system:  beam [originally], beam [presently]
* Frequency:          2000 kHz
* Beams:              3 beams (not attenuated) oriented downward with angle 25 deg to axis
* Transformation matrix::

     1.5774  -0.7891  -0.7891 
      0.000   -1.366    1.366 
     0.3677   0.3677   0.3677 
* Time ranges from 2008-06-25 10:00:00 to 2008-07-07 17:11:40 with 101230 samples and mean increment 10.49798 s
* Data

                             Min.     Mean     Max.   Dim.        OriginalName
    v [m/s]                  -0.5     0.006987 0.513  101230x25x3 -           
    a                        0        96.421   202    101230x25x3 -           
    q                        0        53.094   100    101230x25x3 -           
    distance [m]             0.091764 0.5718   1.0518 25          -           
    pressure [dbar]          0        13.164   19.186 101230      -           
    error                    0        0        0      101230      -           
    temperature [°C, ITS-90] 3.73     9.8345   24.31  101230      -           
    heading [°]              0        240.67   359.9  101230      -           
    pitch [°]                -32.3    -5.8362  33.8   101230      -           
    roll [°]                 -37.9    -7.9391  37.1   101230      -           

* Processing Log
    - 2017-05-06 14:18:55 UTC: `read.oce(file = "/data/archive/sleiwex/2008/moorings/m05/adp/nortek_aqd2843/raw/adp_nortek_aqd2843.prf")`
```

---

![a](https://cloud.githubusercontent.com/assets/99469/25774545/4f55fd58-3267-11e7-907b-77b1a03753db.png)

**Comment by dankelley** [comment]
I'm closing this now, on the hope that @richardsc will take a quick look at some time, opening it again if needs be.

PS. I don't have old graphs of this dataset but I bet @richardsc does ...
